php/Terminus/Models/Collections/TerminusCollection.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Terminus\Models\Collections;
+
+use \TerminusCommand;
+
+abstract class TerminusCollection {
+  protected $fetch_args = array();
+  protected $fetch_url  = '';
+  protected $models     = array();
+  protected $owner_name = '';
+
+  /**
+   * Instantiates the collection, sets param members as properties
+   *
+   * @param [array] $options To be set to $this->key
+   * @return [TerminusCollection] $this
+   */
+  public function __construct($options = array()) {
+    foreach ($options as $key => $option) {
+      $this->{$key} = $option;
+    }
+  }
+
+  /**
+   * Retrieves all models
+   *
+   * @return [array] $this->models
+   */
+  public function all() {
+    if (empty($this->models)) {
+      $this->fetch();
+    }
+    return $this->models;
+  }
+
+  /**
+   * Retrieves the model of the given ID
+   *
+   * @param [string] $id ID of desired model instance
+   * @return [TerminusModel] $this->models[$id]
+   */
+  public function get($id) {
+    $models = $this->all();
+    if (isset($models[$id])) {
+      return $models[$id];
+    }
+    return null;
+  }
+
+  /**
+   * List Model IDs
+   *
+   * @return [array] $ids Array of all model IDs
+   */
+  public function ids() {
+    $models = $this->all();
+    $ids    = array_keys($models);
+    return $ids;
+  }
+
+  /**
+   * Fetches model data from API and instantiates their model instances
+   *
+   * @return [void]
+   */
+  private function fetch() {
+    $model   = $this->getModelName();
+    $owner   = $this->owner_name;
+    $options = array_merge(
+      array('options' => array('method' => 'get')),
+      $this->fetch_args
+    );
+    $results = TerminusCommand::simple_request(
+      $this->fetch_url,
+      $options
+    );
+
+    foreach (get_object_vars($results['data']) as $id => $model_data) {
+      $model_data->id = $id;
+      $options        = array($owner => $this->$owner);
+      $this->models[$id] = new $model(
+        $model_data,
+        $options
+      );
+    }
+  }
+
+  /**
+   * Returns name of the model collected by this collection
+   *
+   * @return [string] $model_name Name of model
+   */
+  private function getModelName() {
+    $name_array = explode('\\', get_class($this));
+    $model_name = $name_array[0]
+      . '\\' . $name_array[1]
+      . '\\' . substr(array_pop($name_array), 0, -1);
```

Comment:
Currently this only removes the -s from the end of the line. All our models' names right now make that doable, but if that should change, we will need [a good inflector](https://packagist.org/packages/doctrine/inflector).